packages/smooth_app/lib/data_models/product_list_supplier.dart

```diff
@@ -25,17 +25,22 @@ abstract class ProductListSupplier {
   /// Returns a helper supplier in order to refresh the data
   ProductListSupplier? getRefreshSupplier();
 
-  /// Clears the database and restarts from top page.
-  Future<void> clear() async {
+  /// Clears the database from page 2.
+  ///
+  /// Use-case: we have data on several pages, and the user wants to launch
+  /// the search again. That means
+  /// * all cached results are deprecated
```

Comment:
```suggestion
  /// * all cached results are now invalid
```

Reply:
@VaiTon The cached results don't become "invalid". They are legitimate results that the user wants to refresh: if the refresh does not work (e.g. server down) we won't show "invalid" results, we will show "not refreshed" results.

Reply:
With "invalid" I was referring to this kind of Invalidation: https://en.wikipedia.org/wiki/Cache_invalidation

Reply:
@VaiTon Again, it's not exactly the same case: I don't _mark_ the data as invalid, in a system that would desperately try to refresh all invalid data asap. It's just a one shot try to refresh the data.
Well, good enough is good enough, if we find a better unambiguous word some day we'll use it!